bgcArgoDMQC/io/index.py

```diff
@@ -25,47 +25,60 @@ def index_exists():
     '''
 
-    url_dir = resource.URL_DIR_DICT[url]
-    if mission == 'B':
-        local_filename = resource.path('Index') / 'argo_bio-profile_index.txt.gz'
-        remote_filename = 'ftp://' + url + (Path(url_dir) / 'argo_bio-profile_index.txt.gz').as_posix()
-    elif mission == 'C':
-        local_filename = resource.path('Index') / 'ar_index_global_prof.txt.gz'
-        remote_filename = 'ftp://' + url + (Path(url_dir) / 'ar_index_global_prof.txt.gz').as_posix()
-    elif mission == 'S':
-        local_filename = resource.path('Index') / 'argo_synthetic-profile_index.txt.gz'
-        remote_filename = 'ftp://' + url + (Path(url_dir) / 'argo_synthetic-profile_index.txt.gz').as_posix()
-    elif mission == 'M':
-        local_filename = resource.path('Index') / 'ar_index_global_meta.txt.gz'
-        remote_filename = 'ftp://' + url + (Path(url_dir) / 'ar_index_global_meta.txt.gz').as_posix()
-    elif mission == 'T':
-        local_filename = resource.path('Index') / 'ar_index_global_traj.txt.gz'
-        remote_filename = 'ftp://' + url + (Path(url_dir) / 'ar_index_global_traj.txt.gz').as_posix()
-    else: # pragma: no cover
-        raise ValueError('Input {} not recognized'.format(mission))
-
-    if remote:
+    if source != 'argopy':
+        warnings.warn('For most use cases loading the index via argopy (source="argopy") is preferred.')
+        url_dir = resource.URL_DIR_DICT[url]
+        if mission == 'bgc-b':
+            local_filename = resource.path('Index') / 'argo_bio-profile_index.txt.gz'
+            remote_filename = 'ftp://' + url + (Path(url_dir) / 'argo_bio-profile_index.txt.gz').as_posix()
+        elif mission == 'core':
+            local_filename = resource.path('Index') / 'ar_index_global_prof.txt.gz'
+            remote_filename = 'ftp://' + url + (Path(url_dir) / 'ar_index_global_prof.txt.gz').as_posix()
+        elif mission == 'bgc-s':
+            local_filename = resource.path('Index') / 'argo_synthetic-profile_index.txt.gz'
+            remote_filename = 'ftp://' + url + (Path(url_dir) / 'argo_synthetic-profile_index.txt.gz').as_posix()
+        elif mission == 'meta':
+            local_filename = resource.path('Index') / 'ar_index_global_meta.txt.gz'
+            remote_filename = 'ftp://' + url + (Path(url_dir) / 'ar_index_global_meta.txt.gz').as_posix()
+        elif mission == 'traj':
+            local_filename = resource.path('Index') / 'ar_index_global_traj.txt.gz'
+            remote_filename = 'ftp://' + url + (Path(url_dir) / 'ar_index_global_traj.txt.gz').as_posix()
+        else: # pragma: no cover
+            raise ValueError(f'Input {mission} not recognized - must be *bgc-b*, *core*, *bgc-s*, *traj*, or *meta*')
+
+    if source == 'argopy':
+        if mission == 'traj' or mission == 'meta':
+            raise ValueError('argopy does not support traj or meta file handling yet - use source="local" or source="remote" to load.')
```

Comment:
@cgrdn  argopy does support the ``ar_index_global_meta.txt`` index file

https://argopy.readthedocs.io/en/latest/advanced-tools/stores/argoindex.html#index-file-supported